src/savegame.cpp

```diff
@@ -1613,9 +1613,8 @@ void game::unserialize_master( std::istream &fin )
 {
     savegame_loading_version = 0;
     chkversion( fin );
-    std::unique_ptr<static_popup> popup;
     if( savegame_loading_version < 11 ) {
-        popup = std::make_unique<static_popup>();
+        std::unique_ptr<static_popup>popup = std::make_unique<static_popup>();
```

Comment:
This actually changes the code behavior. `static_popup` removes itself from ui stack on deconstruction, so moving it into the if branch makes it deconstructs prematurely and won't be redrawn if the screen is invalidated in the subsequent code.

Reply:
It is not used anywhere but inside of if branch and there are no redraw calls outside. Am I missing something?

Reply:
You're right that currently there are no redraw or input code outside, but in the future someone may add code that redraws or takes input and then the popup will disappear before save loading completes.